shotover-proxy/src/transforms/redis/cache.rs

```diff
@@ -45,18 +100,185 @@ impl RedisConfig {
+            }
+        }
+        Ok(messages_redis_request)
+    }
+
+    /// unwraps redis response messages into cassandra messages. It does this by replacing the Cassandra
+    /// request messages with their corresponding Cassandra response messages and returns them.
+    /// Result is either the modified message_cass_request (now response messages) or and CacheableState::Err.
+    fn unwrap_cache_response(
+        &self,
+        messages_redis_response: Messages,
+        mut cassandra_messages: Messages,
+    ) -> Result<Messages, CacheableState> {
+        // Replace cass_request messages with cassandra responses in place.
+        // We reuse the vec like this to save allocations.
+        let mut messages_redis_response_iter = messages_redis_response.into_iter();
+        /* there is a redis response for each statement in a CassandraMessage so we have to map
+        the redis responses back to the cassandra requests
+         */
+
+        for cass_request in cassandra_messages.iter_mut() {
+            // the responses for this request
+            let cassandra_result: Result<CassandraResult, CacheableState> =
+                if let Some(Frame::Cassandra(frame)) = cass_request.frame() {
+                    let queries = frame.operation.queries();
+                    if queries.len() != 1 {
+                        Err(CacheableState::Err {
+                            reason: "Cacheable Cassandra query must be only one statement".into(),
+                        })
+                    } else if let Some(mut redis_response) = messages_redis_response_iter.next() {
+                        match redis_response.frame() {
+                            Some(Frame::Redis(redis_frame)) => {
+                                match redis_frame {
+                                    RedisFrame::SimpleString(_) => Err(CacheableState::Err {
+                                        reason: "Redis returned a simple string".into(),
+                                    }),
+                                    RedisFrame::Error(e) => {
+                                        return Err(CacheableState::Err {
+                                            reason: e.to_string(),
+                                        })
+                                    }
+                                    RedisFrame::Integer(_) => Err(CacheableState::Err {
+                                        reason: "Redis returned an int value".into(),
+                                    }),
+                                    RedisFrame::BulkString(redis_bytes) => {
+                                        // Redis response contains serialized version of result struct from CassandraOperation::Result( result )
+                                        let mut cursor = Cursor::new(redis_bytes.as_ref());
+                                        let answer =
+                                            CassandraResult::from_cursor(&mut cursor, Version::V4);
```

Comment:
We cant just assume its V4.
Fixing it sounds complicated so at the very least add a `TODO: dont hardcode version`